src/Kestrel.Transport.Abstractions/Internal/TransportConnection.cs

```diff
@@ -14,10 +15,15 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions.Internal
     public abstract partial class TransportConnection : ConnectionContext
     {
         private IDictionary<object, object> _items;
+        private List<(Action<object> handler, object state)> _heartbeatHandlers;
+        private readonly object _heartbeatLock = new object();
+        private CancellationTokenSource _connectionClosingCts = new CancellationTokenSource();
```

Comment:
Shouldn't this be disposed? Particularly because it might not fire. I know we dispose our registration, but other connection middleware may not.

Reply:
Also nit: _connectionClosedRequestedCts